test/cases/expressions/xhp-comment.hack

```diff
@@ -0,0 +1,6 @@
+return <frag>
+    <!--  comment
+    one  -->
+    {$this->test()}
```

Comment:
Unrelated to the current fix, but looks like grammar.js isn't handling `xhp_braced_expression` here based on your test. I think it might be due to the fact that `xhp_string` allows the left-brace `{` token. I checked with `hh_parse --full-fidelity-errors-all` and it is indeed not a valid standalone token in XHP expressions.

```
> hh_parse --full-fidelity-errors-all <(echo '<frag></frag>;')
> hh_parse --full-fidelity-errors-all <(echo '<frag>{</frag>;')
(1,1)-(1,16) XHP: mismatched tag: '' not the same as 'frag'
(1,9)-(1,9) An expression is expected here.
(1,15)-(1,15) An expression is expected here.
(1,15)-(1,15) A right brace ('}') is expected here.
(2,1)-(2,1) A closing XHP tag is expected here.
(2,1)-(2,1) A semicolon (';') is expected here.
```